_blog/2023-11-22-web-analytics-rust.mdx

```diff
@@ -0,0 +1,553 @@
+---
+title: "Building a Web Analytics Tool with JS, htmx and Rust" 
+description: This article documents how you can build a simple privacy-respecting analytics tool using a JavaScript script and handle the backend with Rust. 
+author: josh
+tags: [rust, static site generator, frontend, guide]
+thumb: ssg-in-rust-thumb.png 
+cover: ssg-in-rust-thumb.png
+date: '2023-11-15T14:30:00'
+---
+
+htmx is a lightweight JavaScript library for dynamic web development, enabling real-time content updates without full-page reloads. 
+
+This time, we're building a privacy-respecting analytics tool which uses htmx, Tera and Axum with Shuttle.
+
+1) Run `cargo shuttle init --from joshua-mo-143/shuttle-analytics-ex` and follow the prompt
+2) `cd` to the folder
+3) Run `cargo shuttle deploy --allow-dirty` and your static site generator is live!
+
+Don't forget to make sure that `cargo-shuttle` is installed and that you're logged in! You can also visit the repo [here.](https://www.github.com/joshua-mo-143/shuttle-analytics-ex)
+
+## Getting Started
+You'll need to use `cargo shuttle init --template axum` to get started (if you don't have it installed, you can use `cargo install cargo-shuttle` then log in using `cargo shuttle login` to get started!). Once you're done, you'll want to use the following one-liner to install your dependencies:
+
+```bash
+cargo add axum chrono serde serde-json shuttle-metadata shuttle-shared-db sqlx tera tokio url -F \
```

Comment:
might just be better to show the cargo toml contents at this point

Reply:
Hmmmm, yeah. There are a lot of dependencies